test/integration/basic_test.go

```diff
@@ -0,0 +1,112 @@
+// +build integration
+
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/DATA-DOG/godog"
+	"github.com/DATA-DOG/godog/gherkin"
+
+	"github.com/minishift/minishift/test/integration/util"
+)
+
+type Minishift struct {
+	runner util.MinishiftRunner
+}
+
+func (m *Minishift) aPersona(personaName string) error {
+	// not implemented
+	return nil
+}
+
+func (m *Minishift) shouldBeRunning() error {
+	if !m.runner.CheckStatus("Running") {
+		return fmt.Errorf("Expected minishift to be running")
+	}
+
+	return nil
+}
+
+func (m *Minishift) shouldHaveAValidIPAddress() error {
```

Comment:
👍 right, these methods are what I meant with building up a Minishift specific set of helper functions (step definitions) which can be reused by the various test descriptions (feature files)